cli/cli-utils/src/packageIsInstallable.ts

```diff
@@ -20,6 +23,24 @@ export function packageIsInstallable (
   const pnpmVersion = packageManager.name === 'pnpm'
     ? packageManager.stableVersion
     : undefined
+  if (pkg.packageManager) {
+    const [pmName, pmVersion] = pkg.packageManager.split('@')
+    if (pmName && pmName !== 'pnpm') {
+      const msg = `This project is configured to use ${pmName}`
+      if (opts.packageManagerStrict) {
+        throw new PnpmError('OTHER_PM_EXPECTED', msg)
+      } else {
+        globalWarn(msg)
+      }
+    } else if (pmVersion && pnpmVersion && pmVersion !== pnpmVersion) {
+      const msg = `This project is configured to use v${pmVersion} of pnpm. Your current pnpm is v${pnpmVersion}`
```

Comment:
it looks like this will force non-corepack users to use corepack, otherwise, different projects have different requirements and this error will be annoying. (btw I support using corepack)

Reply:
oh I see `config.package-manager-strict=false`

Reply:
Why would it force them to use corepack? They can just install the required version of pnpm. Although, I don't like the strictness of this setting. It would make more sense to require the provided or newer version of pnpm.

Reply:
They'll need to switch pnpm version manually if they're not using corepack, otherwise pnpm will give an error imo.